components/RichText/RichText.tsx

```diff
@@ -0,0 +1,148 @@
+import { render, type StoryblokRichtext } from 'storyblok-rich-text-react-renderer';
+import { cnb } from 'cnbuilder';
+import { SbLink } from '@/components/Storyblok/partials/SbLink';
+import {
+  Heading,
+  Paragraph,
+  Text,
+  textAligns,
+  textColors,
+  textVariants,
+  type TextColorType,
+  type FontSizeType,
+  type TextAlignType,
+} from '@/components/Typography';
+import { wysiwygClasses, type WysiwygClassesType } from '@/utilities/wysiwygClasses';
+
+/**
+ * "default" means using the default body font size (defined in TW base) as the base font size
+ * "card" means using the card-paragraph style as the base font size (smaller than default)
+ */
+export type RichTextBaseFontSizeType = 'default' | 'card' | 'base23';
+
+export type RichTextProps = {
+  wysiwyg: StoryblokRichtext;
+  baseFontSize?: RichTextBaseFontSizeType;
+  textColor?: TextColorType;
+  textAlign?: TextAlignType;
+  className?: string;
+};
+
+export const RichText = ({
+  wysiwyg,
+  baseFontSize = 'default',
+  textColor = 'black',
+  textAlign = 'left',
+  className,
+}: RichTextProps) => {
+  const printColor = 'print:text-black';
+
+  const rendered = render(wysiwyg, {
+    markResolvers: {
+      styled: (children, props) => {
+        // Custom classes are string of class name(s) separated by spaces, e.g., 'ood-has-tab-before su-sans su-before-bg-bay-dark', 'su-bold'
+        const { class: wysiwygCustomClasses = '' } = props;
+
+        // Expanded icon options as needed
+        const hasIcon = wysiwygCustomClasses.includes('su-link--action');
+
+        /**
+         * Split the string of legacy classes into an array, filter out empty strings, and map to the corresponding TW class names
+         * If a match is not found in the wysiwygClasses object, return the original legacy custom class name from the WYSIWYG
+         * Stitch the array back into a string at the end
+         */
+        const finalClasses = wysiwygCustomClasses.split(/\s+/)
+          .filter(Boolean)
+          .map((className) =>
+            className in wysiwygClasses ? wysiwygClasses[className as WysiwygClassesType] : className,
+          ).join(' ');
+
+        return (
+          <Text
+            as="span"
+            icon={hasIcon ? 'chevron-right' : undefined}
+            iconProps={{ className: hasIcon ? 'inline-block ml-03em group-hocus-within:translate-x-02em' : undefined }}
+            className={finalClasses}
+          >
+            {children}
+          </Text>
```

Comment:
In Storyblok, custom classes can be added to a `span` wrapper around some text. Some of these legacy classes addes icons from Decanter 6 - I added the icon prop to the Text component which seems handy here. LMK if you're ok with this approach. I also maps some of these legacy classes eg, `.ood-has-tab-before`, `.su-serif` to the new TW classes.